cmd/kubeadm/app/util/runtime/runtime.go

```diff
@@ -151,20 +151,28 @@ func (runtime *DockerRuntime) RemoveContainers(containers []string) error {
 
 // PullImage pulls the image
 func (runtime *CRIRuntime) PullImage(image string) error {
-	out, err := runtime.exec.Command("crictl", "-r", runtime.criSocket, "pull", image).CombinedOutput()
-	if err != nil {
-		return errors.Wrapf(err, "output: %s, error", string(out))
+	var lastError error
+	for i := 0; i < constants.PullImageRetry; i++ {
+		out, err := runtime.exec.Command("crictl", "-r", runtime.criSocket, "pull", image).CombinedOutput()
+		if err == nil {
+			return nil
+		}
+		lastError = errors.Wrapf(err, "output: %s, error", string(out))
 	}
-	return nil
+	return lastError
```

Comment:
do you need `lastError`?
it seems that you can return directly `return errors.Wrapf(err, "output: %s, error", string(out))`